test/suite/test_compact11.py

```diff
@@ -0,0 +1,190 @@
+        self.pr("Original bitmap in binary: " + orig_bits)
+        self.pr("Reopened bitmap in binary: " + new_bits)
+        for o_bit, n_bit in zip(orig_bits, new_bits):
+            if o_bit != '0':
+               self.assertTrue(n_bit != '0')
+
+    def parse_blkmods(self, uri):
+        meta_cursor = self.session.open_cursor('metadata:')
+        config = meta_cursor[uri]
+        meta_cursor.close()
+        # The search string looks like: ,blocks=feffff1f000000000000000000000000
+        # Obtain just the hex string.
+        b = re.search(',blocks=(\w+)', config)
+        self.assertTrue(b)
+        # The bitmap string after the = is in group 1.
+        blocks = b.group(1)
+        self.pr(f"parse_blkmods for '{uri}': {blocks}")
+        return blocks
+
+    def test_compact11(self):
+        # Create N tables.
+        uris = []
+        files = []
+        bitmaps = []
+        for i in range(self.num_tables):
+            uri = self.uri_prefix + f'_{i}'
+            uris.append(uri)
+            files.append(f'file:test_compact11_{i}.wt')
+            self.session.create(uri, self.create_params)
+
+        # Populate the first half of each table.
+        for uri in uris:
+            self.populate(uri, 0, self.table_numkv // 2, self.value_size)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Take a full backup that will be used for incremental backups later during the test.
+        os.mkdir(self.backup_incr)
+        self.initial_backup = True
+        self.take_full_backup(self.backup_incr)
+        self.initial_backup = False
+
+        # Insert the latter 50% in each table.
+        for uri in uris:
+            self.populate(uri, self.table_numkv // 2, self.table_numkv, self.value_size)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Delete 50% of each file.
+        for uri in uris:
+            self.delete_range(uri, 50 * self.table_numkv // 100)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Get the bitmaps of each file.
+        for uri in files:
+            bitmaps.append(self.parse_blkmods(uri))
+
+        # Turn on background compaction to allow the bitmap blocks to be modified
+        # from compact operation.
+        self.turn_on_bg_compact('free_space_target=1MB')
+        while self.get_files_compacted(uris) < self.num_tables:
+            time.sleep(0.5)
+
+        assert self.get_files_compacted(uris) == self.num_tables
+        self.pr(f'Compaction has compacted {self.num_tables} tables.')
+
+        # Update the incremental backup ID from the parent class.
+        self.bkup_id += 1
+        self.take_incr_backup(self.backup_incr)
+
+        self.pr('Comparing the original bitmaps with the new ones...')
+        for index, uri in enumerate(files):
+            new_bitmap = self.parse_blkmods(uri)
+            self.compare_bitmap(bitmaps[index], new_bitmap)
+
+        # Take a second full backup.
```

Comment:
This extra backup is not needed, we can remove it.

Reply:
Actually we need this backup because we should be using it to compare against the incremental backup. We expect no difference in data. I have moved this to earlier in the test before compaction starts [cc01737](https://github.com/wiredtiger/wiredtiger/pull/9977/commits/cc01737af49a21ded94fdde966dff35ec018d323)